.github/workflows/migrations-metadata-deployment.yml

```diff
@@ -0,0 +1,80 @@
+name: "Applies the migrations to the database"
+
+on:
+  push:
+    branches:
+      - main
+      - production
+      - frank/version-control-ecs-task-definitions
+    paths:
+      - "moped-database/**"
+      - ".github/workflows/migrations-metadata-deployment.yml"
+      - ".github/workflows/migrations-metadata-deployment-helper.sh"
+
+jobs:
+  apply-migrations-and-metadata:
+    name: Apply Migrations and Metadata
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/setup-python@v4
+        name: "Setup Python"
+        with:
+          python-version: "3.8"
+          architecture: "x64"
+      # Get the code first
+      - name: "Checkout Code"
+        uses: actions/checkout@v3
+      # Then install the AWC CLI tools & boto3
+      - name: "Install AWS Cli & Hasura CLI"
+        run: |
+          sudo apt-get install python3-setuptools
+          # reminder; this is V1 of the AWS CLI
+          pip3 install awscli virtualenv
+          curl -L https://github.com/hasura/graphql-engine/raw/stable/cli/get.sh | bash
+
+      # Run the shell commands using the AWS environment variables
+      - name: "Apply Migrations and Metadata"
+        env:
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_MOPED_HASURA_CONFIGURATION_FILE_PRODUCTION: ${{ secrets.AWS_MOPED_HASURA_CONFIGURAITON_FILE_PRODUCTION }}
+          AWS_MOPED_HASURA_CONFIGURATION_FILE_STAGING: ${{ secrets.AWS_MOPED_HASURA_CONFIGURAITON_FILE_STAGING }}
+        run: |
+          export BRANCH_NAME=${GITHUB_REF##*/}
+          echo "SHA: ${GITHUB_SHA}"
+          echo "ACTION/BRANCH_NAME: ${BRANCH_NAME}"
+          echo "GR: ${GITHUB_REF}"
+          echo "PWD: $(pwd)"
+          source $(pwd)/.github/workflows/migrations-metadata-deployment-helper.sh
+          # run_migration_process
+
+  update-ecs-task-deployment:
+    name: Check for updated ECS task definition and deploy if needed
+    runs-on: ubuntu-24.04
+    needs: apply-migrations-and-metadata
+    if: always()
+    steps:
+      - name: "Checkout Code"
+        uses: actions/checkout@v3
+
+      - name: "Install AWS CLI v2"
+        run: |
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          sudo ./aws/install --update
```

Comment:
I was surprised to learn that we use the AWS cli version 1 in our old, existing CI here, but in retrospect, the timing is not too surprising. V2 came out in 2020. V1 is still supported, but V2 is the future.

Anyway, that's why I `--update` here.